Post-condition on function in impl fails if function uses `&mut Self`

```rust
#[hax_lib::attributes]
trait T {
    #[hax_lib::ensures(|_|true)]
    fn v(x:&mut Self);
}

```
[Open this code snippet in the playground](https://hax-playground.cryspen.com/#fstar/3aad8b7f51/gist=1e718c75696600f4589d2d33e8155644)

```
error[E0411]: cannot find type `Self` in this scope
 --> src/lib.rs:4:17
  |
1 | #[hax_lib::attributes]
  | ---------------------- `Self` not allowed in a function
...
4 |     fn v(x:&mut Self);
  |                 ^^^^ `Self` is only available in impls, traits, and type definitions
```

Works if the ensures is removed.

--- a/.github/workflows/test_installs.yml
+++ b/.github/workflows/test_installs.yml
@@ -22,7 +22,7 @@ jobs:
           - ubuntu-latest
           - ubuntu-20.04
           - macos-latest
-          - macos-12
+          - macos-13
     runs-on: ${{ matrix.os }}
     steps:
     - uses: actions/checkout@v3

--- a/hax-lib/macros/src/utils.rs
+++ b/hax-lib/macros/src/utils.rs
@@ -218,8 +218,6 @@ pub fn make_fn_decoration(
     mut generics: Option<Generics>,
     self_type: Option<Type>,
 ) -> (TokenStream, AttrPayload) {
-    let uid = ItemUid::fresh();
-    let mut_ref_inputs = unmut_references_in_inputs(&mut signature);
     let self_ident: Ident = syn::parse_quote! {self_};
     let error = {
         let mut rewriter = RewriteSelf::new(self_ident, self_type);
@@ -230,6 +228,8 @@ pub fn make_fn_decoration(
         }
         rewriter.get_error()
     };
+    let uid = ItemUid::fresh();
+    let mut_ref_inputs = unmut_references_in_inputs(&mut signature);
     let decoration = {
         let decoration_sig = {
             let mut sig = signature.clone();

--- a/tests/attributes/src/lib.rs
+++ b/tests/attributes/src/lib.rs
@@ -381,3 +381,11 @@ mod requires_mut {
         }
     }
 }
+
+mod issue_1266 {
+    #[hax_lib::attributes]
+    trait T {
+        #[hax_lib::ensures(|_|true)]
+        fn v(x: &mut Self);
+    }
+}

